lib/dynamoid/finders.rb

```diff
@@ -36,9 +36,12 @@ def find(*ids)
         ids = Array(ids.flatten.uniq)
         if ids.count == 1
           result = self.find_by_id(ids.first, options)
+          raise Errors::RecordNotFound if result.nil?
           expects_array ? Array(result) : result
         else
-          find_all(ids)
+          result = find_all(ids)
+          raise Errors::RecordNotFound if result.size != ids.size
```

Comment:
Do we want to provide the error message too?

ActiveRecord will say which id was passed and for an array of ids it'll say which ids passed and it'll produce "found X results, but was looking for N" kind of message so we could help report the difference.

Reply:
@richardhsu Done